### DISCUSSION TITLE
`jobs.<job_id>.if` not working?

### DISCUSSION THREAD

**Discussion by arnested** [community/community#24802] [Apps, API and Webhooks]
<p>I have created a workflow with jobs.&lt;job_id&gt;.if but it fails with:</p>
<pre><code>- Your workflow file was invalid: .github/workflows/test.yml (Line: 7, Col: 5): Unexpected value 'if'
</code></pre>
<p><a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idif" rel="nofollow noopener">Documentation</a> says it should be possible.</p>
<p>I created a small example at: <a href="https://github.com/reload/actions-test/pull/1/checks" rel="nofollow noopener">https://github.com/reload/actions-test/pull/1/checks</a></p>
<p><strong>Edit August 14, 2019:</strong>  jobs.&lt;job_id&gt;.if has been removed from the documentation now. See workaround below.</p>
<p><strong>Edit October 1, 2019:</strong> it’s back, working and documented!</p>

**Comment by arnested** [comment]
<p>As a workaround I moved the if condition into each step in the job.</p>
<p>There are some drawbacks to that though.</p>
<ul>
<li>I need to remember the condition on each step in the job</li>
<li>When step uses actions docker containers for each step is still being build and only the execution of the step is being ignored.<br>
This results in time wasted on builds I don’t need. And wasted time is slower feedback loops, wasted money (on a paid plan) and wasted computing resources (global warming, etc).</li>
</ul>

**Comment by juulSme** [comment]
<p>I’m having the same issue, putting the <code>if</code> in each step is a pain in the ass.</p>
<pre><code>jobs:
  deploy:
    name: Publish release
    # this causes errors
    if: github.event.action == 'published'
</code></pre>

**Comment by tsunammis** [comment]
<p>I need also this feature, it works on steps but not at the job level.</p>

**Comment by rodrigonehring** [comment]
<p>A workaround :frowning:</p>
<pre><code>jobs:
  checkEvent:
    runs-on: ubuntu-latest

    steps:
      - name: if published
        if: github.event.action != 'published'
        run: exit 1

  build:
    runs-on: ubuntu-latest
    needs: checkEvent
</code></pre>

**Comment by arnested** [comment]
<p>Nice workaround :slight_smile: I’ll try that. It’s probably better than <code>if</code> on every step. Thank you.</p>

**Comment by morphinewan** [comment]
<p>I have the same problem and have asked github support group for help. And I got their reply just now. It seems jobs doesn’t support if statement right now. And the online document is updated and <strong>jobs.&lt;job_id&gt;.if</strong>  seems to be removed.</p>

**Comment by arnested** [comment]
<p><img src="http://www.quickmeme.com/img/f1/f1fd7b1590fd8530b9006758de4fbe040b5774ee3f265e629e839642e4b46916.jpg" alt=""></p>
<p><span class="mention">@rodrigonehrings</span> workaround must the best we can do then…</p>
<p>Thank you for sharing!</p>

**Comment by arnested** [comment]
<p>Ok, I tried the aproach now and it doesn’t give the result I was hoping for.</p>
<p>Since this workaround will make the check job fail if the condition is met we end up with a failed job on i.e. a Pull Request. I wss aiming at something like neutral / skipped.</p>
<p>So back to if on each step then …</p>

**Comment by letmaik** [comment, marked as answer]
<p>I just tried this and it works as expected. I guess the feature is back but documentation is still missing.</p>

**Comment by arnested** [comment]
<p>Yes! I tried it as well and it worked!</p>
<p>Now let’s hope it stays / gets documented.</p>

**Comment by arnested** [comment]
<p>One caveat / percularity:</p>
<p>If you add jobs.&lt;job_id&gt;.if to all jobs of the workflow and none of them evalute to true none of the actions will be run (as expected) and the workflow/jobs will not show up on a pull request (also a expected). But if you go to the Actions tab you will see teh workflow running and appearing to run forever (doing nothing).</p>

**Comment by arnested** [comment]
<p>And with todays announcement (<a href="https://github.blog/changelog/2019-10-01-github-actions-new-workflow-syntax-features/" rel="nofollow noopener">https://github.blog/changelog/2019-10-01-github-actions-new-workflow-syntax-features/</a>) it is also documented again!</p>

**Comment by jackguoAtJogg** [comment]
<p>For some reasons the jobs dont even show up… Can you share your snippet?</p>
